vars/dockerExecute.groovy

```diff
@@ -340,20 +344,35 @@ private getDockerOptions(Map dockerEnvVars, Map dockerVolumeBind, def dockerOpti
     }
 
     if (dockerOptions) {
-        if (dockerOptions instanceof CharSequence) {
-            dockerOptions = [dockerOptions]
-        }
-        if (dockerOptions instanceof List) {
-            dockerOptions.each { String option ->
-                options << escapeBlanks(option)
-            }
-        } else {
-            throw new IllegalArgumentException("Unexpected type for dockerOptions. Expected was either a list or a string. Actual type was: '${dockerOptions.getClass()}'")
-        }
+        options.addAll(dockerOptionsToList(dockerOptions))
     }
+
     return options.join(' ')
 }
 
+@NonCPS
+def securityContextFromOptions(dockerOptions) {
+    Map securityContext = [:]
+
+    if (!dockerOptions) {
+        return null
+    }
+
+    def userOption = dockerOptionsToList(dockerOptions).find { (it.startsWith("-u ") || it.startsWith("--user ")) }
+    if (!userOption) {
+        return null
+    }
+
+    def userGroupIds = userOption.split(" ")[1].split(":")
```

Comment:
Let's make sure that there is a second part of this option.

Reply:
done